### DISCUSSION TITLE
Preventing manual triggers (without config)

### DISCUSSION THREAD

**Discussion by mumblemaker** [apache/airflow#23245] [Q&A]
Our DAGs can be categorised as either:
1. Safe to run at any time
2. Should only be run based on schedule (data changed based on date variables)

Am I right that there is currently no way to prevent someone triggering a manual DAG run from the GUI without setting the config (specifically date/time) first? I couldn't find anything relevant on the DAG documents (https://airflow.apache.org/docs/apache-airflow/stable/concepts/dags.html).

**Comment by msumit** [comment]
Probably you can use DAG Params
<https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html> to
accept a mandatory date parameter and validate it against some range?
